### PR TITLE
Read expressions from a file

### PR DESCRIPTION
**Pull request by fariedt**
Adds the `::read` command to the console.

### CODE REVIEW

File: src/repl/session.rs [hunk 1]
```diff
@@ -625,6 +627,26 @@ impl Session {
...
+
+        let snippet = match fs::read_to_string(filename) {
+            Ok(snippet) => snippet,
+            Err(err) => return output.push(red!(format!("unable to read {}: {}", filename, err))),
+        };
+
+        let mut result = match self.formatted_interpretation(snippet, None, false, None, None) {
+            Ok((output, _)) => output,
+            Err(result) => result,
+        };
```
Comment (obycode) [inline src/repl/session.rs:645]: After reading from the file, I think it should be treated the same as when a snippet is passed in. The missing piece here then would be handling when `show_costs` mode is enabled.

Perhaps a clean solution would be to refactor the `snippet => {...}` case of the match in `handle_command` into a function, then call that function here.

**Review by obycode — CHANGES_REQUESTED** [review]
Thanks for adding this @fariedt! I just have one change request.

File: src/repl/session.rs (hunk 1 above)
Comment (fariedt) [reply src/repl/session.rs]: I've pushed a commit for this.  It's a clean change; I was able to merge both `handle_command` and the cli `get_costs` but not the non-cli `get_costs`.  The main change in behavior from my first commit is that `::read` can now show the execution costs if the user has toggled cost reporting.

**Review by obycode — CHANGES_REQUESTED** [review]
Thanks for this change. I approved, but now I see that the wasm build is failing because of that change. I guess we do need two versions of `run_snippet` for the cli and non-cli modes. That should then allow you to get merge `get_costs` completely into these new methods.

File: src/repl/session.rs (hunk 1 above)
Comment (obycode) [reply src/repl/session.rs]: See new review comment.

File: src/repl/session.rs [hunk 2]
```diff
@@ -1202,6 +1217,12 @@ impl Session {
         output.append(&mut result);
     }
 
+    #[cfg(not(feature = "cli"))]
+    pub fn get_costs(&mut self, output: &mut Vec<String>, cmd: &str) {
+        let snippet = cmd.to_string().split_off("::get_costs ".len());
+        self.run_snippet(output, true, &snippet.to_string());
+    }
```
Comment (obycode) [inline src/repl/session.rs:1224]: Looks like the two versions of `get_costs` are identical now, so we can merge into one and remove the feature gating.

**Review by obycode — CHANGES_REQUESTED** [review]

**Review by obycode — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**